spec/services/diaper_partner_client_spec.rb

```diff
@@ -29,6 +29,26 @@
       result = DiaperPartnerClient.get(id: 123)
       expect(result).to eq 'success'
     end
+
+    context 'with query_params' do
+      let(:query_params) { { additional_details: true } }
+
+      it 'performs a GET request' do
+        stub_partner_request(:get, 'https://partner-register.com/123?additional_details=true')
+        result = DiaperPartnerClient.get(id: 123, query_params: query_params)
+        expect(result).to eq 'success'
```

Comment:
One way to make the tests even more robust is by randomly generating the id you pass into here. We have the really awesome [faker gem](https://github.com/faker-ruby/faker) that generates random data. So instead you could do:
```
fake_random_id = Faker::Number.number
stub_partner_request(:get, "https://partner-register.com/#{fake_random_id}?additional_details=true")
result = DiaperPartnerClient.get(id: fake_random_id, query_params: query_params)
```

This is a minor change request :) And more letting you know this is possible